Source/Patches.cs

```diff
@@ -542,4 +1044,117 @@ public static void Postfix()
 			}
 		}
 	}
-}
\ No newline at end of file
+
+	[HarmonyPatch(typeof(UIRoot), nameof(UIRoot.UIRootUpdate))]
```

Comment:
Extra note: Patching the same method over and over isn't very effectiv. At least patch it once and call all the postfixes with their own method name in a single postfix. That also means that you only need a single try/catch which speeds things up and adds the error handling to all your uirootupdate code.

Reply:
I agree, was thinking of doing this - but wasn't sure if that was best.  One thing is different patches might need to happen at different times though.  I could refactor this to introduce a couple patches and combine all functions into ones that fit their times perhaps?